rust/arrow/src/array/cast.rs

```diff
@@ -40,12 +40,20 @@ where
         .expect("Unable to downcast to dictionary array")
 }
 
-pub fn as_list_array<S: OffsetSizeTrait>(arr: &ArrayRef) -> &GenericListArray<S> {
+pub fn as_generic_list_array<S: OffsetSizeTrait>(arr: &ArrayRef) -> &GenericListArray<S> {
     arr.as_any()
         .downcast_ref::<GenericListArray<S>>()
         .expect("Unable to downcast to list array")
 }
 
+pub fn as_list_array(arr: &ArrayRef) -> &ListArray {
```

Comment:
Not sure whether we need an extra public function for a one-liner.

Reply:
@jorgecarleitao the main reason for adding that is to keep these two cast function names consistent with others. i.e. for a particular `FooBarArray` type, there exists a `as_foo_bar_array` cast function. However, if you have strong opinion on this, I am happy to remove it.

I have added the inline annotation.

Reply:
Makes sense. Let's keep it 👍